### PR TITLE
Move and rename attach metadata

### PR DESCRIPTION
**Pull request by callumforrester**
Fixes #ISSUE

### Instructions to reviewer on how to test:
1. Confirm tests pass

### Checks for reviewer
- [ ] Would the PR title make sense to a scientist on a set of release notes
- [ ] If a new device has been added does it follow the [standards](https://github.com/DiamondLightSource/dodal/wiki/Device-Standards)
- [ ] If changing the API for a pre-existing device, ensure that any beamlines using this device have updated their Bluesky plans accordingly
- [ ] Have the connection tests for the relevant beamline(s) been run via `dodal connect ${BEAMLINE}`

### CODE REVIEW

File: src/dodal/plans/data_session_metadata.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+from dodal.common.beamlines import beamline_utils
+from dodal.common.types import MsgGenerator, UpdatingDirectoryProvider
+
+DATA_SESSION = "data_session"
+DATA_GROUPS = "data_groups"
+
+
+def attach_data_session_metadata_wrapper(
+    plan: MsgGenerator, provider: UpdatingDirectoryProvider | None
```
Comment (callumforrester) [inline src/dodal/plans/data_session_metadata.py:14]:
```suggestion
    plan: MsgGenerator, provider: UpdatingDirectoryProvider | None = None
```

**Comment by stan-dot** [comment]
@callumforrester does this need to be tested on the beamline before merging?

File: src/dodal/plans/data_session_metadata.py (hunk 1 above)
Comment (stan-dot) [reply src/dodal/plans/data_session_metadata.py]: done

**Comment by callumforrester** [comment]
@stan-dot No, an approval from someone either than the two of us is all that's needed

**Review by DiamondJoseph — APPROVED** [review]

**Review by DiamondJoseph — APPROVED** [review]

**Pull request merged**